extensions/amp-form/0.1/amp-form.js

```diff
@@ -209,39 +221,70 @@ export class AmpForm {
       this.setState_(FormState_.SUBMITTING);
       const isHeadOrGet = this.method_ == 'GET' || this.method_ == 'HEAD';
       let xhrUrl = this.xhrAction_;
-      if (isHeadOrGet) {
-        xhrUrl = addParamsToUrl(this.xhrAction_, this.getFormAsObject_());
+
+      const varSubPromises = [];
+      for (let i = 0; i < varSubsFields.length; i++) {
+        const variable = varSubsFields[i].getAttribute('default-value');
+        varSubPromises.push(
+            this.urlReplacement_.expandAsync(variable).then(value => {
```

Comment:
Careful, `UrlReplacements` isn't a generic variable substitution library. I added an assert that the "protocol" of the url remains the same (because they were all URLs before now), but that doesn't apply here. So if the default value had something like `RANDOM:X:Y`, this would break because of the changing "protocol".

We should split UrlReplacements into a core class (`VarSubstitutions`?), and leave the URL specific code out of it.